src/treemodel/selection.js

```diff
@@ -135,8 +156,44 @@ export default class Selection {
 	}
 
 	/**
+	 * Returns the first range in the selection. First range is the one which {@link core.treeModel.Range#start start} position
```

Comment:
1. Some time ago we discussed whether ranges should be sorted upon adding. If yes, we would have to change `focus` and `anchor` implementation. Live ranges are also an issue. Most of time there will be one range in selection anyway so I don't think that sorting on-the-fly will be that difficult. What we could do is sort upon adding and sort ranges whenever it's live ranges changes but I suspect that this might happen more often or as often as using `.getFirstRange()`. The other idea is to introduce some kind `._dirty` flag that will be checked whenveer `.getFirstRange()` is fired and if selection is not dirty, there won't be any sorting. But this doesn't seem good to me either.

2. We cannot require that ranges are added in the right order. Multiple ranges are for multiple selection and multiple selection might be done in any order.